### PR TITLE
 ESP32 examples Readme.md update with chip-tool instructions

### PR DESCRIPTION
**Pull request by jadhavrohit924**
#### Problem
What is being fixed?  Examples:
* Fix  ESP32 examples Readme with chip-tool instructions

#### Change overview
changed ESP32 examples Readme with chip-tool instructions

#### Testing
How was this tested? (at least one bullet point required)
* No testing required

### CODE REVIEW

File: examples/all-clusters-app/esp32/README.md [hunk 1]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
-    application and is configurable through menuconfig.
+    $ chip-tool pairing ble-wifi ${NODE_ID_TO_ASSIGN} ${SSID} ${PASSWORD} 0 20202021 3840
 
-         - chip-device-ctrl > ble-scan
-         - chip-device-ctrl > connect -ble 3840 20202021 135246
+or equivalently:
 
-         Parameters:
-         1. Discriminator: 3840 (configurable through menuconfig)
-         2. Setup-pin-code: 20202021 (configurable through menuconfig)
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:188]: You can keep pt. 1 and 2 above

File: examples/all-clusters-app/esp32/README.md [hunk 2]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
-         3. Node ID: Optional.
-            If not passed in this command, then it is auto-generated by the controller and displayed in the output of connect.
-            The same value should be used in the next commands.
-            We have chosen a random node ID which is 135246.
+    $ chip-tool pairing ble-wifi 17 hex:787980 hex:736563726574 0 20202021 3840
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:194]: What is this? If not required, please remove

File: examples/all-clusters-app/esp32/README.md [hunk 3]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
-    it using `EnableWiFiNetwork` command. In this example, we have used
-    `TESTSSID` and `TESTPASSWD` as the SSID and passphrase respectively.
+#### Pair a device over IP
 
-         - chip-device-ctrl > zcl NetworkCommissioning AddOrUpdateWiFiNetwork 135246 0 0 ssid=str:TESTSSID credentials=str:TESTPASSWD breadcrumb=0
+The command below will discover devices with long discriminator 3840 and try to
+pair with the first one it discovers using the provided setup code.
 
-         - chip-device-ctrl > zcl NetworkCommissioning ConnectNetwork 135246 0 0 networkID=str:TESTSSID breadcrumb=0
+    $ chip-tool pairing onnetwork-long ${NODE_ID_TO_ASSIGN} 20202021 3840
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:201]: This seems incorrect. Please remove

File: examples/all-clusters-app/esp32/README.md [hunk 4]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
+Run the built executable and pass it the discriminator and pairing code of the
+remote device, as well as the network credentials to use.
 
-### Commissioning over BLE
+The command below uses the default values hard-coded into the debug versions of
+the ESP32 all-clusters-app to commission it onto a Wi-Fi network:
 
--   Establish the secure session over BLE. BLE is the default mode in the
-    application and is configurable through menuconfig.
+    $ chip-tool pairing ble-wifi ${NODE_ID_TO_ASSIGN} ${SSID} ${PASSWORD} 0 20202021 3840
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:190]: 1. You have mentioned `out/debug/standalone/chip-tool` above and directly use `chip-tool` in the commands at all the places
2. The 0 in the above command is not required now

File: examples/all-clusters-app/esp32/README.md [hunk 5]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
-    `chip-device-ctrl > zcl OnOff Off 135246 1 1`
+To use the Client to send Matter commands, run the built executable and pass it
+the target cluster name, the target command name as well as an endpoint id.
 
--   Use the LevelControl cluster commands to control the CurrentLevel attribute.
-    This allows you to control the brightness of the led.
+The endpoint id must be between 1 and 240.
 
-    `chip-device-ctrl > zcl LevelControl MoveToLevel 135246 1 1 level=10 transitionTime=0 optionMask=0 optionOverride=0`
+    $ chip-tool onoff on 12344321 1
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:221]: You have used `${NODE_ID_TO_ASSIGN}` above and directly use 12344321 here

File: examples/all-clusters-app/esp32/README.md [hunk 6]
```diff
@@ -160,73 +174,60 @@ your network configuration. To erase it, simply run.
...
-    color of on-board LED.
+The client will send a single command packet and then exit.
 
-    `zcl ColorControl MoveToHue 135246 1 1 hue=100 direction=0 transitionTime=0 optionsMask=0 optionsOverride=0`
-    `zcl ColorControl MoveToSaturation 135245 1 1 saturation=200 transitionTime=0 optionsMask=0 optionsOverride=0`
+### levelcontrol
+
+```bash
+Usage:
+  ./out/debug/chip-tool levelcontrol move-to-level Level=10 TransitionTime=0 OptionMask=0 OptionOverride=0  12344321 1
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:229]: Now this path again `./out/debug/chip-tool` is different from the one mentioned

File: examples/bridge-app/esp32/README.md [hunk 7]
```diff
@@ -151,6 +151,20 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
 
           $ idf.py -p /dev/tty.SLAB_USBtoUART monitor
 
+## Setting up chip-tool
```
Comment (dhrishi) [inline examples/bridge-app/esp32/README.md:154]: Please make similar changes in other examples as well

File: examples/all-clusters-app/esp32/README.md (hunk 1 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md:188]: done.

File: examples/all-clusters-app/esp32/README.md (hunk 2 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

File: examples/all-clusters-app/esp32/README.md (hunk 3 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

File: examples/all-clusters-app/esp32/README.md (hunk 4 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: resolved

File: examples/all-clusters-app/esp32/README.md (hunk 5 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

File: examples/all-clusters-app/esp32/README.md (hunk 6 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

File: examples/bridge-app/esp32/README.md (hunk 7 above)
Comment (jadhavrohit924) [reply examples/bridge-app/esp32/README.md]: done

File: examples/all-clusters-app/esp32/README.md [hunk 8]
```diff
@@ -160,73 +160,67 @@ your network configuration. To erase it, simply run.
 
           $ idf.py -p /dev/tty.SLAB_USBtoUART erase_flash
 
-### Setting up Python Controller
+### Setting up chip-tool
 
-Once ESP32 is up and running, we need to set up a device controller to perform
-commissioning and cluster control.
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/README.md:166]: I suggest to keep this line

**Review by dhrishi — APPROVED** [review]

File: examples/all-clusters-app/esp32/README.md (hunk 8 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md:166]: done.

File: examples/all-clusters-app/esp32/README.md [hunk 9]
```diff
@@ -160,73 +160,71 @@ your network configuration. To erase it, simply run.
 
           $ idf.py -p /dev/tty.SLAB_USBtoUART erase_flash
 
-### Setting up Python Controller
+-   Once ESP32 is up and running, we need to set up a device controller to
+    perform
+-   commissioning and cluster control.
```
Comment (shubhamdp) [inline examples/all-clusters-app/esp32/README.md:165]: This line is supposed to go with the above bullet point, in the below files as we.

File: examples/all-clusters-app/esp32/README.md [hunk 10]
```diff
@@ -160,73 +160,71 @@ your network configuration. To erase it, simply run.
...
+$ cd examples/chip-tool
+
+$ rm -rf out
 
-           $ source ./out/python_env/bin/activate
-           $ chip-device-ctrl
+$ git submodule update --init
 
-### Commissioning over BLE
+$ source third_party/connectedhomeip/scripts/activate.sh
```
Comment (shubhamdp) [inline examples/all-clusters-app/esp32/README.md:181]: This action is already listed in setting up the `Setting up the environment` section, also in the below file.

File: examples/all-clusters-app/esp32/README.md (hunk 9 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

File: examples/all-clusters-app/esp32/README.md (hunk 10 above)
Comment (jadhavrohit924) [reply examples/all-clusters-app/esp32/README.md]: done

**Review by electrocucaracha — APPROVED** [review]

**Review by shubhamdp — APPROVED** [review]

**Comment by SgtPepperFTW** [comment]
What about old example with python controller? It is not valid anymore? What about commissioning the wifi credentials and resolving the IPV6 with chip-tool? How is that done?

**Comment by dhrishi** [comment]
@SgtPepperFTW Python controller should work as well. But, as the default testing tool now for certification is chip-tool, we've moved the documentation to mention chip-tool.

The single command `pairing ble-wifi`, as mentioned above, will do Wi-Fi provisioning as well as resolving the IP address. With chip-tool, it is done with a single command instead of 3 to 4 commands required for Python controller.

**Comment by SgtPepperFTW** [comment]
@dhrishi I had problem with Python controller, cause I didn't manage to resolve the NodeID with it. Not only it finished with "Couldn't resolve...", but it also crashed with "memory access violation"

My other question is, does each "virtual" Light get a dedicated unicast IPv6 address? My understanding is each "end-device" in Matter network is supposed to have unique IPv6 device. Or does it go only for real Wifi and Thread devices?

To the chip tool: Can I retrieve a list of all bridge-app ESP32 that were commissioned on the network? Also, can I get a list of all end-point NodeId with description of their clusters assigned to such ESP32s?

**Comment by PSONALl** [comment]
> I had problem with Python controller, cause I didn't manage to resolve the NodeID with it. Not only it finished with "Couldn't resolve...", but it also crashed with "memory access violation"
@SgtPepperFTW can you share the commit id on which you are trying this

**Comment by PSONALl** [comment]
> My other question is, does each "virtual" Light get a dedicated unicast IPv6 address? My understanding is each "end-device" in Matter network is supposed to have unique IPv6 device. Or does it go only for real Wifi and Thread devices?
Only real Wifi and Thread devices will have unique IPv6 address

**Comment by PSONALl** [comment]
> To the chip tool: Can I retrieve a list of all bridge-app ESP32 that were commissioned on the network? Also, can I get a list of all end-point NodeId with description of their clusters assigned to such ESP32s?

You can get list of device types, endpoints and clusters using Descriptor cluster

**Comment by SgtPepperFTW** [comment]
@PSONALl it was commit 99c1985ad229dec6198d623a54101ca1656afa43

I figured out that I can use Descriptor cluster, but I wasn't able to retrieve all the information I wanted.

The example main.cpp contains 
static Device gLight1("Light 1", "Office");
static Device gLight2("Light 2", "Office");
static Device gLight3("Light 3", "Kitchen");
static Device gLight4("Light 4", "Den");

But descriptor doesn't return, for example, "Office" or "Light 1", when I am commanding ./chip-tool descriptor read device-list 12344321 2.

Also it returns Type 256 for Lights and Type 14 on Node-Id 0. I understand 256 probably means OnOff type and Type 14 means "Bridge App", but I can't find in source code, where does it come from and what other types there exist.

**Pull request merged**